feat: migration guide for v0.76 (#1238)
SigNoz/signoz.io: docs/userguide/send_logs_http.md

---
title: Sending Logs to SigNoz over HTTP
id: send_logs_http.md
---

You can send your logs to SigNoz over HTTP.

> The paylod is an array of json logs. It follows the same structure as [OTEL Logs Data Model](https://opentelemetry.io/docs/specs/otel/logs/data-model/), here is how the payload looks like.

Author: ```suggestion
The payload is an array of JSON logs. It follows the same structure as
[OTEL Logs Data Model](https://opentelemetry.io/docs/specs/otel/logs/data-model/).

Here is how the payload looks like:
```

> ```

Author: ```suggestion
```json
```

[
  {
    "timestamp": <uint64>,
    "trace_id": <hex string>,
    "span_id": <hex string>,
    "trace_flags": <int>
    "severity_text": <string>,
    "severity_number": <int>,
    "attributes": <map>,
    "resources": <map>,
    "body": <string>,
  },
  ...
]
```

Note :- 
  * `timestamp` is int64 nanoseconds since Unix epoch
  * You can use `message` instead of `body`.
  * Any other keys present apart from the ones mentioned aboved will be moved to the attributes map.
    Ex:- 
      ```json
      [
        {
          "host": "myhost",
          "method": "GET",
          "body": "this is a log line"

>         },

Author: ```suggestion
        }
```

Using valid JSON format.

      ]
      ```

      Will be finally treated as 
      ```json
      [
        {
          "attributes": {
            "host": "myhost",

>             "method": "GET",

Author: ```suggestion
            "method": "GET"
```

same as above

          },
          "body": "this is a log line"

>         },

Author: ```suggestion
        }
```

same as above

      ]
      ```

## Send logs to SigNoz in SigNoz cloud

* Here is a sample curl request

>   ```

Author: ```suggestion
  ```bash
```

  curl --location 'https://ingest.<REGION>.signoz.cloud/logs/json/' \
  --header 'Content-Type: application/json' \
  --header 'signoz-access-token: <SIGNOZ_INGESTION_KEY>' \
  --data '[
      {
          "timestamp": 1698310066000000000,
          "trace_id": "000000000000000018c51935df0b93b9",
          "span_id": "18c51935df0b93b9",
          "trace_flags": 0,
          "severity_text": "info",
          "severity_number": 4,
          "attributes": {
              "method": "GET",
              "path": "/api/users"
          },
          "resources": {
              "host": "myhost",
              "namespace": "prod"
          },
          "message": "This is a log line"
      }
  ]'
  ```

  `<SIGNOZ_INGESTION_KEY>` is the ingestion key.
  

>   Depending on the choice of your region for SigNoz cloud, the otlp endpoint will vary according to this table.

Author: ```suggestion
  `<REGION>` is the name of the region.

  Depending on the choice of your region for SigNoz Cloud,
  the OTLP endpoint will vary according to the table below:
```

## Send logs to SigNoz in Self-Hosted SigNoz

* Modify the `docker-compose.yaml` file present inside `deploy/docker/clickhouse-setup` to expose a port, in this case `8081`.
    ```yaml {8}
    ...
    otel-collector:
        image: signoz/signoz-otel-collector:0.79.12
        command: ["--config=/etc/otel-collector-config.yaml"]
        volumes:
          - ./otel-collector-config.yaml:/etc/otel-collector-config.yaml
        ports:
          - "8082:8082"
    ...
    ```

* Add the httpreceiver reciever to `otel-collector-config.yaml` which is present inside `deploy/docker/clickhouse-setup`
    ```yaml {2-10}
    receivers:
      httpreceiver/json:
        endpoint: 0.0.0.0:8082
        source: json
    ...
    ```

* Next we will modify our pipeline inside `otel-collector-config.yaml` to include the receiver we have created above.
    ```yaml {4}
    service:
        ....
        logs:
            receivers: [otlp, httpreceiver/json]
            processors: [batch]
            exporters: [clickhouselogsexporter]
    ```

* Now we can restart the otel collector container so that new changes are applied and we can send our logs to port `8082`.

* Sample curl

>   ```

Author: ```suggestion
  ```bash
```

>   curl --location 'localhost:8082' \

Author: @nityanandagohain have you tested this curl locally?

Author: Add a note below to say to replace `localhost` with the IP of the instance/VM.

> --header 'Content-Type: application/json' \
> --data '[

Author: ```suggestion
  --header 'Content-Type: application/json' \
  --data '[
```

---
title: Sending Logs to SigNoz over HTTP

> id: send_logs_http.md

Reviewer: ```suggestion
id: send-logs-http.md
```

@nityanandagohain we follow this style mostly for links in blog and docs, so would be good to have it for logs too.

Reviewer 2: Cool, have updated it for this doc.

---

> You can send your logs to SigNoz over HTTP.

Reviewer: @nityanandagohain so we have decided to add some overview in each doc to give readers a bit of context when they land up on any page in docs as an entry point. Check out the [hostmetrics guide](https://signoz.io/docs/userguide/hostmetrics/) for reference. 

You will be the best judge here. This might not need an overview but I wanted to share. The idea is to give the user an expectation of what will he accomplish by following this guide.

Reviewer 2: Don't feel anything else is required right now.

The payload is an array of JSON logs. It follows the same structure as
[OTEL Logs Data Model](https://opentelemetry.io/docs/specs/otel/logs/data-model/).

Here is how the payload looks like:
```
[
  {
    "timestamp": <uint64>,
    "trace_id": <hex string>,
    "span_id": <hex string>,
    "trace_flags": <int>
    "severity_text": <string>,
    "severity_number": <int>,
    "attributes": <map>,
    "resources": <map>,
    "body": <string>,
  }
]
```

Note :- 
* `timestamp` is int64 nanoseconds since Unix epoch
* You can use `message` instead of `body`.

> * Any other keys present apart from the ones mentioned aboved will be moved to the attributes map.

Reviewer: ```suggestion
* Any other keys present apart from the ones mentioned above will be moved to the attributes map.
```

  Ex:- 
  
  ```json
  [
    {
      "host": "myhost",
      "method": "GET",
      "body": "this is a log line"
    }
  ]
  ```

>   Will be finally treated as 

Reviewer: ```suggestion
  Will be finally treated as:
```

  ```json
  [
    {
      "attributes": {
        "host": "myhost",
        "method": "GET"
      },
      "body": "this is a log line"
    }
  ]
  ```

## Send logs to SigNoz in SigNoz cloud

> * Here is a sample curl request

Reviewer: ```suggestion
* Here is a sample curl request:
```

  ```bash

>   curl --location 'https://ingest.<REGION>.signoz.cloud/logs/json/' \

Reviewer: ```suggestion
  curl --location 'https://ingest.<REGION>.signoz.cloud/logs/json/' \
```

Region won't have any endpoint like the ones we have in APM: https://ingest.{region}.signoz.cloud:443, just confirming.

Reviewer 2: 443 is the default one. It doesn't matter. but let me add that so that it is consistant.

  --header 'Content-Type: application/json' \
  --header 'signoz-access-token: <SIGNOZ_INGESTION_KEY>' \
  --data '[
      {
          "timestamp": 1698310066000000000,
          "trace_id": "000000000000000018c51935df0b93b9",
          "span_id": "18c51935df0b93b9",
          "trace_flags": 0,
          "severity_text": "info",
          "severity_number": 4,
          "attributes": {
              "method": "GET",
              "path": "/api/users"
          },
          "resources": {
              "host": "myhost",
              "namespace": "prod"
          },
          "message": "This is a log line"
      }
  ]'
  ```

  `<SIGNOZ_INGESTION_KEY>` is the ingestion key.

  `<REGION>` is the name of the region.
  

>   Depending on the choice of your region for SigNoz Cloud, the OTLP endpoint will vary according to the table below:

Reviewer: @nityanandagohain not sure how should the user use the below link. I see in the config above that location has this url: `https://ingest.<REGION>.signoz.cloud/logs/json/`

So will the final url become: https://ingest.in.signoz.cloud:443/logs/json/. If yes, we should include `443` in the config file itself as it is common.

  | Region | Endpoint                   |
  | ------ | -------------------------- |
  | US     | ingest.us.signoz.cloud:443 |
  | IN     | ingest.in.signoz.cloud:443 |
  | EU     | ingest.eu.signoz.cloud:443 |

> * Once you run the above curl, you can open your SigNoz UI to verify it.

Reviewer: @nityanandagohain can we share here what will be shown in SigNoz UI - like which tab or what data. Would be great if we can have a screenshot here.

## Send logs to SigNoz in Self-Hosted SigNoz

* Modify the `docker-compose.yaml` file present inside `deploy/docker/clickhouse-setup` to expose a port, in this case `8081`.
    ```yaml {8}
    ...
    otel-collector:
        image: signoz/signoz-otel-collector:0.79.12
        command: ["--config=/etc/otel-collector-config.yaml"]
        volumes:
          - ./otel-collector-config.yaml:/etc/otel-collector-config.yaml
        ports:

>           - "8082:8082"

Reviewer: ```suggestion
          - "8082:8082"
```

We have written in the description above `in this case 8081`, in the config it is 8082. Is this intended?

    ...
    ```

> * Add the httpreceiver reciever to `otel-collector-config.yaml` which is present inside `deploy/docker/clickhouse-setup`

Reviewer: ```suggestion
* Add the httpreceiver reciever to `otel-collector-config.yaml` which is present inside `deploy/docker/clickhouse-setup`:
```

    ```yaml {2-10}
    receivers:
      httpreceiver/json:
        endpoint: 0.0.0.0:8082
        source: json
    ...
    ```

> * Next we will modify our pipeline inside `otel-collector-config.yaml` to include the receiver we have created above.

Reviewer: @Calm-Rock is there any best practice for giving out instructions which requires multiple changes in the same config file. The config file here is long.
